internal/provider/resource_database.go

```diff
@@ -338,6 +322,161 @@ func resourceDatabaseDelete(ctx context.Context, d *schema.ResourceData, meta in
 	return nil
 }
 
+func resourceDatabaseUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*astra.ClientWithResponses)
+	databaseID := d.Id()
+	cloudProvider := d.Get("cloud_provider").(string)
+
+	if d.HasChangeExcept("region") {
+		// only region changes supported at the moment
+		return diag.Errorf("Update of Database resource only supported for fields: %s", "region")
+	}
+
+	if d.HasChange("region") {
+		// get regions to add and delete
+		regionsToAdd, regionsToDelete := getRegionUpdates(d.GetChange("region"))
+		if len(regionsToAdd) > 0 {
+			// add any regions to add first
+			if err := addRegionsToDatabase(ctx, d, client, regionsToAdd, databaseID, cloudProvider); err != nil {
+				return err
+			}
+		}
+		if len(regionsToDelete) > 0 {
+			// delete any regions that should be removed
+			if err := deleteRegionsFromDatabase(ctx, d, client, regionsToDelete, databaseID, cloudProvider); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getRegionUpdates(oldRegions interface{}, newRegions interface{}) ([]string, []string){
```

Comment:
Should we should consider some unit tests for the regions updates?  They seem relatively decoupled from integration with other resources.

Reply:
Good idea. I'll add some.